examples/header-group.html

```diff
@@ -0,0 +1,89 @@
+<template id=header-group>
+
+<style>
+
+  header-group
+    { display: block }
+
+  header-group
+    hgroup
+      { background: #bada55; color: var(--darkest-color) }
+
+</style>
+
+<hgroup>
+  <h1><a href=/examples/index.html>snuggsi ツ</a></h1>
+  <slot name=header><h2>try an example</h2></slot>
+</hgroup>
+
+</template>
+
+<script>
+const tag = `header-group`
+
+Element `header-group`
+
+(class extends HTMLElement {
+
+  static onimportload (event) {
+    let template = event.target.import
+      .querySelector ('template')
+
+      document
+        .querySelector('head')
+        .appendChild (template)
+
+      // possibly should be adopting node into document
+      // and should do it on DOMContentLoaded hook up onimportload
+
+    const shadow = function (element) {
+      let fragment = template.content.cloneNode (true)
+
+      fragment.slots =
+        Array.from (fragment.querySelectorAll ('slot'))
+
+      element.slots =
+        Array.from (element.querySelectorAll ('[slot]'))
+
+      element.slots.map (function (namedslot) {
+        fragment.slots
+          .filter (slot =>
+            (slot.getAttribute ('name') === namedslot.getAttribute ('slot'))
+          )
+          .map (slot =>
+            // Replace slot due to uncertainty
+            // https://github.com/devpunks/snuggsi/pull/23/files#r113990636
+
+            // prefer to use replaceWith however support is sparse
+            // https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/replaceWith
+            // using `Node.parentNode` & `Node.replaceChid` as is defined in (ancient) W3C DOM Level 1,2,3
+            // https://developer.mozilla.org/en-US/docs/Web/API/Node/parentNode
+            // https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild
+
+            slot.parentNode
+              .replaceChild (namedslot, slot)
```

Comment:
@brandondees Fixed in https://github.com/devpunks/snuggsi/pull/23/commits/fb5b6a0ea04b7711eed9acbb36f043ce6405f62e

![capture d ecran 2017-04-28 a 14 38 26](https://cloud.githubusercontent.com/assets/38223/25542437/9d769666-2c20-11e7-90b2-f1af5a7e23b3.png)